docs/annotations.md: add list of known annotations

unixwitch opened the pull request:
Per discussion in #770 and on Slack, create a list of known annotations across
various Ingress controller implementations.  This will (hopefully) increase
compatibility across implementations and reduce the need for users to change
all their annotations when changing Ingress controllers.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/ingress/776)
<!-- Reviewable:end -->

aledbf commented:
@jcmoraisjr please review this PR

unixwitch commented:
I've signed the CLA.

jcmoraisjr commented on docs/annotations.md:
```diff
@@ -0,0 +1,73 @@
+* `nginx`: the `kubernetes/ingress` nginx controller
+* `gce`: the `kubernetes/ingress` GCE controller
+* `traefik`: Traefik's built-in Ingress controller 
+* `haproxy`: Joao Morais' [haproxy controller](https://github.com/jcmoraisjr/haproxy-ingress)
```
Please use `[HAProxy Ingress controller]` instead

jcmoraisjr commented on docs/annotations.md:
```diff
@@ -0,0 +1,73 @@
+| Name | Meaning
+| --- | ---
+| `ssl-passthrough` | Pass TLS connections directly to backend; do not offload.  Default `false`.  (nginx)
+| `ssl-redirect` | Redirect non-TLS requests to TLS when TLS is enabled.  Default `true`.  (nginx, trafficserver)
```
haproxy also implement `ssl-redirect`

jcmoraisjr commented on docs/annotations.md:
```diff
@@ -0,0 +1,73 @@
+
+| Name | Meaning
+| --- | ---
+| `app-root` | Redirect requests outside this path prefix to this prefix. (nginx, trafficserver)
```
`app-root`, actually, "Configures the path to redirect to when the root path `/` of the domain is requested from the client" but with a better english then mine. haproxy also implements.

unixwitch commented:
> app-root, actually, "Configures the path to redirect to when the root path / of the domain is requested from the client"

I've fixed that (and I'll change the TS implementation to match), but this is really unclear in the [nginx documentation](https://github.com/kubernetes/ingress/blob/master/controllers/nginx/configuration.md#rewrite):

> If the Application Root is exposed in a different path and needs to be redirected, the annotation ingress.kubernetes.io/app-root might be used.

I'll make a separate PR to clean that up.

I've fixed the other two comments.

aledbf commented:
/lgtm

aledbf commented:
Merging. Only docs

aledbf commented:
@unixwitch thanks!
